addons/mail/static/src/web/composer_patch.js

```diff
@@ -7,7 +7,13 @@ import { patch } from "@web/core/utils/patch";
 patch(Composer.prototype, "web", {
     get placeholder() {
         if (this.thread && this.thread.model !== "discuss.channel") {
-            if (this.props.type === "message") {
+            if (
+                this.thread.type === "chatter" &&
+                !this.thread.hasWriteAccess &&
+                !(this.thread.hasReadAccess && this.thread.canPostOnReadonly)
+            ) {
+                return _t("You cannot post on this Document...");
```

Comment:
Would be more appreciated to have a string that explains a bit more clearly why the user cannot post on this document.
E.g. `You do not have the rights to post on this Document...`